databricks/koalas/indexes.py

```diff
@@ -2220,20 +2220,20 @@ def intersection(self, other) -> "Index":
         if isinstance(other, DataFrame):
             raise ValueError("Index data must be 1-dimensional")
         elif isinstance(other, MultiIndex):
-            # Always returns an empty MultiIndex if `other` is MultiIndex.
-            return other.to_frame().head(0).index
+            # Always returns a no-named empty Index if `other` is MultiIndex.
+            return self._kdf.head(0).index.rename(None)
         elif isinstance(other, Index):
             spark_frame_other = other.to_frame().to_spark()
             keep_name = self.name == other.name
         elif isinstance(other, Series):
             spark_frame_other = other.to_frame().to_spark()
-            keep_name = self.name == other.name
+            keep_name = True
```

Comment:
Now `Index.intersection` always keeps the name of `Index` when `other` is `Series` from pandas 1.2.